src/logging/monitor.ts

```diff
@@ -113,12 +166,45 @@ class LogMonitor {
 			JOURNALCTL_ERROR_RETRY_DELAY_MAX,
 		);
 		log.debug(
-			`Spawning another process to watch balena.service logs in ${
-				wait / 1000
-			}s`,
+			`Spawning another process to watch host service logs in ${wait / 1000}s`,
 		);
 		await setTimeout(wait);
-		void this.start();
+		void this.start(logSystemMessage);
+	}
+
+	private async getJournalctlOptions(): Promise<SpawnJournalctlOpts> {
+		try {
+			// On SV start, journalctl is spawned with a timestamp to only
+			// get logs since the last Supervisor State.FinishedAt. This will catch any
+			// host and container logs generated while the Supervisor was not running.
+			const supervisorContainer = await getSupervisorContainer();
+			if (supervisorContainer !== null) {
+				const finishedAt = supervisorContainer.State.FinishedAt;
+				const finishedAtDate = new Date(finishedAt).getTime();
+				// When a container has never exited with any exit code,
+				// the FinishedAt timestamp is "0001-01-01T00:00:00Z". Any
+				// timestamp below 0 in ms value is invalid, so use the process
+				// start time instead.
+				if (finishedAtDate > 0) {
+					this.lastSentTimestamp = finishedAtDate;
```

Comment:
Are you sure this works? What happens if the underlying `journalctl` process dies? If I'm reading this correctly the `start` method will be called again and it will repeat logs from before the supervisor started, right?

Reply:
Good catch, I modified the `lastSentTimestamp` calculation logic to only return the last FinishedAt timestamp of the Supervisor if the lastSentTimestamp is null (i.e. this is the first time logMonitor has been started). Changes [here](https://github.com/balena-os/balena-supervisor/pull/2382/commits/2ffac417ba363671da98546360fe6495b6fa1e28#diff-ba81f47a86a6a2d1ec1033fcc6d92c30d75d3557278a367721609771c55f4662R156-R199)